### PR TITLE
Avoid duplicate Pokemon

### PR DESCRIPTION
**Pull request by M4RC0Sx**
Check if a Pokemon is already tracked before adding it.

### CODE REVIEW

File: forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java [hunk 1]
```diff
@@ -45,7 +45,13 @@ public static void addTrackedEntities(PixelmonEntity pixelmon) {
 
             List<EntityData> entities = TRACKED_ENTITIES.computeIfAbsent(entry.getValue().getName().toLowerCase(), ___ -> Lists.newArrayList());
 
-            entities.add(0, EntityData.of(pixelmon));
+
+            EntityData e = EntityData.of(pixelmon);
+
+            // Check if an Entity with the same UUID is already being tracked using lambda
```
Comment (danorris709) [inline forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java]: Don't add inline comments. If the code isn't clear then it needs to be refactored into something that makes it clearer

File: forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java [hunk 2]
```diff
@@ -45,7 +45,13 @@ public static void addTrackedEntities(PixelmonEntity pixelmon) {
...
+            if (!entities.stream().anyMatch(entityData -> entityData.getEntityUUID().equals(e.getEntityUUID()))) {
```
Comment (danorris709) [inline forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java:79]: Please don't use streams to check like this. Can you refactor this into it's own method and use a for loop.

Also, can you use Objects#equals for checking the UUIDs

File: forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java [hunk 3]
```diff
@@ -56,7 +62,7 @@ public static void addTrackedEntities(PixelmonEntity pixelmon) {
     public static void catchPokemon(PixelmonEntity pixelmon, ServerPlayerEntity catcher) {
         for (List<EntityData> value : TRACKED_ENTITIES.values()) {
             for (EntityData entityData : value) {
-                if (entityData.getEntityUUID() == pixelmon.getUUID()) {
+                if (entityData.getEntityUUID().equals(pixelmon.getUUID())) {
```
Comment (danorris709) [inline forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java:94]: Can you also use Objects#equals here too

File: forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java [hunk 4]
```diff
@@ -71,7 +77,7 @@ public static void defeatPokemon(PixelmonEntity pixelmon, ServerPlayerEntity cat
                     continue;
                 }
 
-                if (entityData.getEntityUUID() == pixelmon.getUUID()) {
+                if (entityData.getEntityUUID().equals(pixelmon.getUUID())) {
```
Comment (danorris709) [inline forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java:109]: Can you also use Objects#equals here too

File: forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java (hunk 2 above)
Comment (M4RC0Sx) [reply forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java:79]: Thanks for your reply! Is there a reason to not use streams? Like performance or something like that.

Comment (danorris709) [reply forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java:79]: Yep, pretty much. Streams are notorious for being bad for performance and generally harder to read (as your comment shows, just reading a for loop or a function name, is far easier

**Comment by danorris709** [comment]
Closing for inactivity

**Pull request closed without merging**